internal/resolver/dns/dns_resolver_test.go

```diff
@@ -764,103 +764,113 @@ func TestDNSResolverExponentialBackoff(t *testing.T) {
 		return t
 	}
 	tests := []struct {
+		name     string
 		target   string
 		addrWant []resolver.Address
 		scWant   string
 	}{
 		{
+			"happy-case-default-port",
 			"foo.bar.com",
 			[]resolver.Address{{Addr: "1.2.3.4" + colonDefaultPort}, {Addr: "5.6.7.8" + colonDefaultPort}},
 			generateSC("foo.bar.com"),
 		},
 		{
+			"happy-case-specified-port",
 			"foo.bar.com:1234",
 			[]resolver.Address{{Addr: "1.2.3.4:1234"}, {Addr: "5.6.7.8:1234"}},
 			generateSC("foo.bar.com"),
 		},
 		{
+			"happy-case-another-default-port",
 			"srv.ipv4.single.fake",
 			[]resolver.Address{{Addr: "2.4.6.8" + colonDefaultPort}},
 			generateSC("srv.ipv4.single.fake"),
 		},
 	}
-	for _, a := range tests {
-		b := NewBuilder()
-		cc := &testClientConn{target: a.target}
-		// Cause ClientConn to return an error.
-		cc.updateStateErr = balancer.ErrBadResolverState
-		r, err := b.Build(resolver.Target{Endpoint: a.target}, cc, resolver.BuildOptions{})
-		if err != nil {
-			t.Fatalf("%v\n", err)
-		}
-		var state resolver.State
-		var cnt int
-		for i := 0; i < 2000; i++ {
-			state, cnt = cc.getState()
-			if cnt > 0 {
-				break
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			b := NewBuilder()
+			cc := &testClientConn{target: test.target}
+			// Cause ClientConn to return an error.
+			cc.updateStateErr = balancer.ErrBadResolverState
+			r, err := b.Build(resolver.Target{Endpoint: test.target}, cc, resolver.BuildOptions{})
+			if err != nil {
+				t.Fatalf("%v\n", err)
```

Comment:
No `\n` is needed; logging type functions like this always include a newline, unlike `fmt.Printf`.  And it's a good idea to print what was happening as well: `t.Fatalf("Error building balancer for target %v: %v", test.target, err)`

Reply:
This was ctrl c and ctrl v from other tests now that I think about it. But yeah, will keep in mind thanks.

Reply:
Changed logging message.

Reply:
Changed.

Reply:
This is exactly the problem with copy-paste programming.  :)  Everyone does it, but make sure you understand and try to scrutinize/code-review anything you copy/paste.